content/tutorials/esc-version-tagging/index.md

```diff
@@ -0,0 +1,254 @@
+
+To get started, use the ESC CLI to create a new project and environment in the organization of your choice. (We'll use the `tutorials` organization in this tutorial as an example.) Name the new project `messages` and the environment `dev`:
+
+```bash
+$ esc env init tutorials/messages/dev
+
+Environment created: tutorials/messages/dev
+```
+
+Once it's created, you'll be able to see the new project in Pulumi Cloud as well by navigating to Pulumi ESC &gt; Environments:
+
+![A screenshow showing a newly created ESC environment](./new-env.png)
+
+Next, you'll create your first setting.
+
+## Create a new configuration setting
+
+Use `esc env set` to create a new setting for the message of the day and add it to the `dev` environment. Name the setting `motd`:
+
+```bash
+$ esc env set tutorials/messages/dev motd 'Hello, world!'
+```
+
+You're now able to use this setting in any way you like. Try using `esc env get`, for example, in combination with `echo` to fetch the message and print it to the terminal:
+
+```bash
+$ echo "$(esc env get tutorials/messages/dev motd --value json)"
+
+"Hello, world!"
+```
+
+## Change the setting's value, produce a new version
+
+Every change to a configuration value in an ESC environment produces a new, immutable snapshot of the environment. These snapshots are called [_versions_](/docs/esc/environments/versioning/) and they're indexed numerically.
+
+As of now, your `dev` environment has two versions:  version `1`, which is empty (every environment starts out this way) and version `2`, which contains a single setting (for `motd`). You can see this reflected in the Pulumi Cloud console as well after clicking into the `dev` environment and navigating to the Versions tab:
+
+![A screenshot showing two revisions of the MOTD setting](./two-revisions.png)
+
+Now let's produce a third version by setting the value of `motd` again:
+
+```bash
+$ esc env set tutorials/messages/dev motd 'Good day, world!'
+
+$ echo "$(esc env get tutorials/messages/dev motd --value json)"
+
+"Good day, world!"
+```
+
+With `esc env version history`, you can confirm that version `3` now exists:
+
+```bash
+$ esc env version history tutorials/messages/dev
+
+revision 3 (tag: latest)
+Author: Christian Nunciato <cnunciato>
+Date:   2024-09-18 15:23:58.997 -0700 PDT
+
+revision 2
+Author: Christian Nunciato <cnunciato>
+Date:   2024-09-18 15:06:54.347 -0700 PDT
+
+revision 1
+Author: Christian Nunciato <cnunciato>
+Date:   2024-09-18 15:04:20.07 -0700 PDT
+```
+
+## Version tagging
+
+In the output above, you may also have noticed that version `3` is now [_tagged_](https://www.pulumi.com/docs/esc/environments/versioning/#tagging-versions) with a label of `latest`. Every environment has a built-in `latest` that always points to the most recent version of the environment. You can fetch an environment by tag by passing an `@` symbol followed by the tag name --- for example, to pull the `latest` version explicitly:
```

Comment:
"... by passing an `@` symbol followed by the tag name --- for example, to pull the `latest` version explicitly:" ->
"... by adding an `@` symbol followed by the tag name to the end of the environment name. For example, to pull the `latest` version:"